### PR TITLE
Improved AWS and Azure discoverer

### PR DESCRIPTION
**Pull request by oxisto**
This PR improves the discoverers, mainly for AWS and Azure. It introduces discovery of AWS network interfaces and volumes.

### CODE REVIEW

**Comment by oxisto** [comment]
Blocked by #821

**Comment by anatheka** [comment]
@oxisto I have fixed the tests and changed the generation for the storage blob and files client. If this fits, I can do the same for the others in Azure.

**Comment by anatheka** [comment]
I have added the missing labels to K8S, except the label for the volume assigned to a pod. The volume label we only get directly from the related storage. The question is, if that is than the right label or if we should use the pod label instead.

**Comment by oxisto** [comment]
> @oxisto I have fixed the tests and changed the generation for the storage blob and files client. If this fits, I can do the same for the others in Azure.

Ah good! The tests still seem to fail because I moved some of the parts related to Azure Disks from storage to compute and now the tests cannot find the necessary mocks.

**Comment by anatheka** [comment]
> @oxisto I have fixed the tests and changed the generation for the storage blob and files client. If this fits, I can do the same for the others in Azure.

Should I refactor the Azure client generation for compute and network as well?

File: service/discovery/aws/compute.go [hunk 1]
```diff
@@ -253,11 +364,45 @@ func (*computeDiscovery) getNameOfVM(vm *typesEC2.Instance) string {
...
+		if aws.ToString(tag.Key) == "Name" {
+			return aws.ToString(tag.Value)
+		}
+	}
+
+	// If no tag with 'name' was found, return instanceId instead
+	return aws.ToString(vm.VolumeId)
+}
+
+// getNameOfVolume returns the name if exists (i.e. a tag with key 'name' exists), otherwise instance ID is used
```
Comment (oxisto) [inline service/discovery/aws/compute.go:379]: wrong documentation text

File: service/discovery/aws/compute.go [hunk 2]
```diff
@@ -253,11 +364,45 @@ func (*computeDiscovery) getNameOfVM(vm *typesEC2.Instance) string {
 	return aws.ToString(vm.InstanceId)
 }
 
-// addARNToVM generates the ARN of a VM instance
-func (d computeDiscovery) addARNToVM(vm *typesEC2.Instance) voc.ResourceID {
+// getNameOfVolume returns the name if exists (i.e. a tag with key 'name' exists), otherwise instance ID is used
+func (*computeDiscovery) getNameOfVolume(vm *typesEC2.Volume) string {
```
Comment (oxisto) [inline service/discovery/aws/compute.go:368]: this is probably absolute now that we have `nameOrId`

File: service/discovery/azure/azure.go [hunk 3]
```diff
@@ -82,6 +85,12 @@ type azureDiscovery struct {
 	cred                azcore.TokenCredential
 	clientOptions       arm.ClientOptions
 	discovererComponent string
+	client              client
+}
+
+type client struct {
```
Comment (oxisto) [inline service/discovery/azure/azure.go:91]: I would suggest `clients`

File: service/discovery/azure/azure.go [hunk 4]
```diff
@@ -82,6 +85,12 @@ type azureDiscovery struct {
 	cred                azcore.TokenCredential
 	clientOptions       arm.ClientOptions
 	discovererComponent string
+	client              client
```
Comment (oxisto) [inline service/discovery/azure/azure.go:88]: `clients`

File: service/discovery/azure/storage.go [hunk 5]
```diff
@@ -523,3 +378,35 @@ func accountName(id string) string {
 	splitName := strings.Split(id, "/")
 	return splitName[8]
 }
+
+// initializeBlobContainerClient creates the client if not already exists
+func (d *azureStorageDiscovery) initializeBlobContainerClient() (err error) {
```
Comment (oxisto) [inline service/discovery/azure/storage.go:383]: I think we just used `initXXX` somewhere else

File: service/discovery/azure/storage.go [hunk 6]
```diff
@@ -523,3 +378,35 @@ func accountName(id string) string {
...
+		err = fmt.Errorf("could not get new blob containers client: %w", err)
+		log.Debug(err)
+		return err
+	}
+
+	return
+}
+
+// initializeFileStorageClient creates the client if not already exists
+func (d *azureStorageDiscovery) initializeFileStorageClient() (err error) {
```
Comment (oxisto) [inline service/discovery/azure/storage.go:399]: same as above

File: service/discovery/discovery.go [hunk 7]
```diff
@@ -69,6 +69,16 @@ type grpcTarget struct {
 	opts   []grpc.DialOption
 }
 
+const DiscoveryEventTypeDiscovererStart = 0
+const DiscoveryEventTypeDiscovererFinished = 1
+
+type DiscoveryEvent struct {
```
Comment (oxisto) [inline service/discovery/discovery.go:75]: missing documentation

**Review by oxisto — COMMENTED** [review]
Some review comments, some of which I should probably address myself :)

File: service/discovery/azure/storage.go (hunk 5 above)
Comment (anatheka) [reply service/discovery/azure/storage.go]: You mean the naming or do you wanted to use the other generic function? 

Is it possible to change these methods to generic methods?

Comment (oxisto) [reply service/discovery/azure/storage.go]: I was referring to the naming. Would be interesting if these could be generic functions.

**Review by anatheka — APPROVED** [review]
:+1:

**Pull request merged**